Start columnar filtering (with @dberzano)

dberzano opened the pull request.

alibuild commented:
1a2fbbb287ba5bbb6f8dde65a200ed052c568b6b: approved: will be automatically merged on successful tests

alibuild commented:
Error while checking build/AliPhysics/release for 1a2fbbb287ba5bbb6f8dde65a200ed052c568b6b:
```
sw/BUILD/AliPhysics-latest/log
[ 40%] Built target libEventMixing.rootmap
[ 40%] Built target libJETAN.rootmap
[ 40%] Built target FASTJETAN
[ 40%] Built target AliHLTJET
[ 40%] Generating libOADB.rootmap
[ 40%] Built target libTender.rootmap
[ 40%] Built target libOADB.rootmap
[ 47%] Built target PWGJEEMCALJetTasks-object
[ 47%] Built target libPWGEMCALtriggerPart.rootmap
[ 47%] Built target libPWGflowBase.rootmap
[ 48%] Built target PWGTools
[ 48%] Built target libPWGGlauber.rootmap
[ 48%] Built target libPWGmftmuondepCharmonium.rootmap
[ 48%] Built target libPWGTools.rootmap
[ 48%] Built target libPWGCALO.rootmap
[ 48%] Built target libPWGTRD.rootmap
[ 48%] Built target libPWGHMTF.rootmap
[ 48%] Built target libPWGCocktail.rootmap
[ 48%] Linking CXX shared library ../../lib/libPWGLFforward2.so
[ 48%] Linking CXX shared library ../../../../lib/libPWGCFCorrelationsJCORRAN.so
[ 48%] Linking CXX shared library ../../lib/libPWGCFebye.so
[ 52%] Built target PWGLFforward2
[ 54%] Built target PWGCFCorrelationsJCORRAN
[ 54%] Built target libPWGCFChaoticity.rootmap
[ 54%] Built target PWGUDbase
[ 54%] Linking CXX shared library ../../lib/libPWGLFcommonutils.so
[ 54%] Linking CXX shared library ../../../lib/libPWGCFfemtoscopy.so
[ 57%] Built target PWGCFebye
[ 57%] Linking CXX shared library ../../../lib/libPWGCFHBTwrtEP.so
[ 57%] Built target PWGLFcommonutils
[ 57%] Built target libPWGCFHCascadeAnalysis.rootmap
[ 57%] Built target libPWGCFK0Analysis.rootmap
[ 57%] Built target libPWGCFunicor.rootmap
[ 57%] Built target libPWGCFV0LamAnalysis.rootmap
[ 57%] Built target PWGGAUtils
Error: Symbol default is not defined in current scope  /mnt/mesos/sandbox/sandbox/sw/SOURCES/AliPhysics/build_AliPhysics_release/0/RUN3/AliAnalysisTaskAO2Dconverter.h:20:
Error: Symbol delete is not defined in current scope  /mnt/mesos/sandbox/sandbox/sw/SOURCES/AliPhysics/build_AliPhysics_release/0/RUN3/AliAnalysisTaskAO2Dconverter.h:24:
[ 57%] Built target PWGCFHBTwrtEP
Error: Symbol delete is not defined in current scope  /mnt/mesos/sandbox/sandbox/sw/SOURCES/AliPhysics/build_AliPhysics_release/0/RUN3/AliAnalysisTaskAO2Dconverter.h:25:
Warning: Error occurred during reading source files
Warning: Error occurred during dictionary source generation
!!!Removing /mnt/mesos/sandbox/sandbox/sw/BUILD/9413106795247ba0a29e557a17421d3c4fa871f1/AliPhysics/RUN3/G__RUN3.cxx /mnt/mesos/sandbox/sandbox/sw/BUILD/9413106795247ba0a29e557a17421d3c4fa871f1/AliPhysics/RUN3/G__RUN3.h !!!
Error: /mnt/mesos/sandbox/sandbox/sw/slc7_x86-64/ROOT/v5-34-30-alice10-7/bin/rootcint: error loading headers...
gmake[2]: *** [RUN3/G__RUN3.cxx] Error 1
gmake[1]: *** [RUN3/CMakeFiles/RUN3-object.dir/all] Error 2
gmake[1]: *** Waiting for unfinished jobs....
[ 57%] Built target libPWGCFKPAnalysis.rootmap
[ 57%] Built target libPWGCFFEMTOSCOPYAOD.rootmap
[ 60%] Built target PWGCFfemtoscopy
```
Full log [here](https://ali-ci.cern.ch/repo/logs/alisw/AliPhysics/7938/1a2fbbb287ba5bbb6f8dde65a200ed052c568b6b/build_AliPhysics_release/fullLog.txt).

dberzano commented:
@pzhristov: this is the conversion of current ESDs to the prototype columnar format for Run 3 you did in a separate macro.

Of course this is absolutely not the final version; it is meant to allow us to collaborate by editing it.

We put it (with @mpuccio) into an analysis task and did the following relevant modifications:

* Converted to analysis task
* Added vertex information and centrality
* Defined a unique "event id" (different trees, i.e. tables, can be joined on this number for instance)
* Added a TTree flush every 1000 events (configurable)

cc: @njacazio

alibuild commented:
6da41266ce642e55ec4d474dc4addbe7c186e8f4: approved: will be automatically merged on successful tests

alibuild commented:
e3037401aef91d1c9a5278e70e56a764d0fd153d: approved: will be automatically merged on successful tests
